src/agent/src/mount.rs

```diff
@@ -1112,6 +1140,40 @@ mod tests {
         skip_if_not_root, skip_loop_by_user, skip_loop_if_not_root, skip_loop_if_root,
     };
 
+    #[test]
+    // Override get_mounts during tests
+    fn get_mounts() -> Result<String, std::io::Error> {
```

Comment:
Can we override `get_mounts()` like this?

Reply:
Override is the wrong terminology. Old C/C++ hacker here; I feel guilty :) 

The `get_mounts` here is in the scope of the test module and hence in its namespace. When the tests are not being run, the test module is not included, and the "real" `get_mounts` is available and used. 
When tests are run, the test's `get_mounts` shadows the original function.

Reply:
I did not want to overcomplicate things; we could also use a trait and mock that trait for testing.

Reply:
Are you sure that this is working as expected? I tried a bit in rust play ground and it does not seem to work:
https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=3bac5b90a52ff856320116e1ddf76277

Reply:
@bergwolf Try this: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=df63af81ac9aab1fef5dfa076f8ba06d

If you do `Run` you will see a full output of a `/proc/mounts` file on the system where the code is executing. 

If you do `Test` you will see the smaller output that is specified in the `mod tets`. 

I added a ` std::assert!(false);` causing a failure, which will trigger `cargo test` to print out stdout, otherwise its suppressed, and the tool does not let me add flags to the `cargo test` command.